code/ecdsap256sha2/Hacl.Impl.ECDSA.P256SHA256.Verification.fst

```diff
@@ -456,13 +456,20 @@ val ecdsa_verification_u8:
     (requires fun h -> live h pubKey /\ live h r /\ live h s /\ live h m)
     (ensures fun h0 result h1 ->
       assert_norm (pow2 32 < pow2 61);
-      let publicKeyX = nat_from_intseq_le (Hacl.Spec.ECDSA.changeEndian (uints_from_bytes_be (as_seq h1 (gsub pubKey (size 0) (size 32))))) in
-      let publicKeyY = nat_from_intseq_le (Hacl.Spec.ECDSA.changeEndian (uints_from_bytes_be (as_seq h1 (gsub pubKey (size 32) (size 32))))) in
-      let r = nat_from_intseq_le (Hacl.Spec.ECDSA.changeEndian (uints_from_bytes_be (as_seq h1 r))) in
-      let s = nat_from_intseq_le (Hacl.Spec.ECDSA.changeEndian (uints_from_bytes_be (as_seq h1 s))) in
+      let publicKeyX = nat_from_bytes_be (as_seq h1 (gsub pubKey (size 0) (size 32))) in
+      let publicKeyY = nat_from_intseq_be (as_seq h1 (gsub pubKey (size 32) (size 32))) in
+      let r = nat_from_intseq_be (as_seq h1 r) in
+      let s = nat_from_intseq_be (as_seq h1 s) in
       modifies0 h0 h1 /\
       result == Hacl.Spec.ECDSA.ecdsa_verification (publicKeyX, publicKeyY) r s (v mLen) (as_seq h0 m))
 
+(* absent in ByteSequence for some reason - does it hold?*) 
+
+assume val uints_from_bytes_be_nat_lemma: #t:inttype{unsigned t /\ ~(U1? t)} -> #l:secrecy_level -> #len:size_nat{len * numbytes t < pow2 32}
+  -> b:lbytes_l l (len * numbytes t) ->
+  Lemma (nat_from_intseq_be (uints_from_bytes_be #t #l #len b) == nat_from_bytes_be b)
```

Comment:
I believe this is readily provable from lemmas in `Lib.ByteSequence`.

Reply:
Done